Ensure the clamav-freshclam SystemD service runs on Fedora 34

jsf9k opened the pull request:
## 🗣 Description ##

This pull requests modifies this Ansible role to ensure that the `clamav-freshclam` SystemD service is enabled for Fedora 34.

## 💭 Motivation and context ##

@dav3r noticed this morning that `freshclam` had not been run on one of our FreeIPA instances since it was deployed about a month ago.  This turned out to be because Fedora 34 switched from running `freshclam` via a `cron` job to running it via a SystemD service just like Debian does.

Resolves #31.

## 🧪 Testing ##

All `pre-commit` hooks and `molecule` tests pass.

## ✅ Checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

jsf9k committed:
Add a tasks file for Fedora 34, and tweak the Ansible code to use it

Fedora is a special case because the ansible_distribution_release is
empty for all versions since 21, so in this case we have to use
ansible_distribution_version to differentiate between versions of the
Fedora OS.

jsf9k committed:
Use a systemd-enabled Docker image for Fedora 34

jsf9k committed:
Adapt tests for Fedora 34 changes

jsf9k committed:
Add the clamav and freshclam service names to OS-specific vars files

jsf9k committed:
Use service name variables

This is necessary since the Fedora 34 task file is symlinked to the
Debian task file; the latter must know the names of the services since
they differ between Debian and Fedora.

jsf9k committed:
Tweak a comment to make it more correct

jsf9k committed:
Remove task that creates the /var/log/clamav directory

I don't remember why this task was added before, but is no longer
necessary.

dav3r commented on tasks/main.yml:
```diff
@@ -21,12 +21,25 @@
     dest: /etc/cron.weekly/virus_scan
     mode: 0755
 
+# This directory is created by Debian, but not be Fedora, for instance
```
Typo:
```suggestion
# This directory is created by Debian, but not by Fedora, for instance
```

dav3r reviewed (approved):
I love it! 🗜️

jsf9k committed:
Add our standard Fedora foo to ensure systemd initializes

In this case, though, we only apply it to Fedora 34 since that is the
only systemd-enabled Fedora Docker image we are using.

mcdonnnj commented on molecule/default/prepare.yml:
```diff
@@ -1,4 +1,31 @@
 ---
+# When installing packages during later steps, the Fedora Docker
+# images we are using (geerlingguy/docker-fedora32-ansible:latest and
+# cisagov/docker-fedora33-ansible:latest) can throw sporadic errors
+# like:
```
Just a thought for accuracy in this project.

```suggestion
# image we are using (geerlingguy/docker-fedora34-ansible:latest) can
# throw sporadic errors like:
```

mcdonnnj commented on molecule/tests/test_default.py:
```diff
@@ -45,13 +43,22 @@ def test_files_and_dirs(host, path):
-        host.system_info.distribution == "debian"
-        or host.system_info.distribution == "ubuntu"
-    ):
+    if host.system_info.distribution in ["debian", "ubuntu"]:
```
Should this be in here?

```suggestion
    if host.system_info.distribution in ["debian", "kali", "ubuntu"]:
```

mcdonnnj reviewed (approved):
Strong work figuring this out. 💪💪💪 I had a comment suggestion and a question about a test for your consideration.

jsf9k committed:
Improve accuracy of comment

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Include Kali in Debian test

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>
